specs/agents/configuration.md

```diff
@@ -10,6 +10,53 @@ Here's a list of the config options across all agents, their types, default valu
 
 They are provided as environment variables but depending on the language there might be several feasible ways to let the user tweak them. For example besides the environment variable `ELASTIC_APM_SERVER_URL`, the Node.js Agent might also allow the user to configure the server URL via a config option named `serverUrl`, while the Python Agent might also allow the user to configure it via a config option named `server_url`.
 
+### Configuration Source Precedence
+
+Configuration can be provided via a number of sources. Values from central
+configuration MUST have the highest precedence, and default values MUST have
+the lowest precedence. Otherwise, agents SHOULD adopt the following config
+source precendence. Sources higher on this list will override values provided
+by sources lower on this list:
+
+ - Central configuration
+ - Environment variables
+ - Inline configuration in code
+ - Config files
```

Comment:
Like central configuration, in the Java agent configs are made dynamic through config files, which means that they must have higher precedence than env or code. 

I would also prefer `MAY`, or even leave the order of the middle options undefined, but we can also use `SHOULD` and not follow it in the Java agent as well.